src/cfg/cfg-traversal.h

```diff
@@ -293,15 +293,36 @@ struct CFGWalker : public PostWalker<SubType, VisitorType> {
     }
   }
 
+  // Create a new basic block and link the current one to continue directly to
+  // it, with no other basic blocks relevant. That is, this is called when we
+  // are in block A and need to create a new block B with A -> B and A has no
+  // other branches from it, and B has no other branches to it, aside from A
+  // possibly exiting the entire function (e.g. via a throw). Some users of this
+  // class can override this behavior to not create a basic block here, if they
+  // don't need to preserve the property that a basic block ends with an
+  // instruction that might branch (that is, if we don't create a new basic
+  // block here then we might transfer control out of the entire function from
+  // the middle of the block).
+  void continueToNewBasicBlock() {
```

Comment:
Nit: How about `continueToUniqueSuccessor` or something? Without reading the comment, this name can be confusing, because we basically continue to a new basic block every time we create a new basic block.

Reply:
Yeah, good point. I've also been thinking about @tlively concerns from earlier about the complexity.

Perhaps it would be simpler to formulate this as "ignores branches to outside of the function." That would apply to all the relevant passes since they just look at locals (which go away anyhow if we branch out). How does that sound @aheejin @tlively ?

Reply:
That sounds good to me. For the analysis framework we will an option like that to configure the correct behavior when building intraprocedural and interprocedural CFGs.

Reply:
Ok, I made that change. I think it is simpler that way, PTAL.

Btw @aheejin - great test in `test/lit/passes/rse-eh.wast`, it caught my misunderstanding here and made it very easy to figure out!

Reply:
Sounds good to me too!